cupy/sorting/search.py

```diff
@@ -24,7 +25,28 @@ def argmax(a, axis=None, dtype=None, out=None, keepdims=False):
     return a.argmax(axis=axis, dtype=dtype, out=out, keepdims=keepdims)
 
 
-# TODO(okuta): Implement nanargmax
+def nanargmax(a, axis=None):
+    """Return the indices of the maximum values in the specified axis ignoring
+    NaNs. For all-NaN slice ``ValueError`` is raised.
+    Subclass cannot be passed yet, subok=True still unsupported
+    Args:
+        a (cupy.ndarray): Array to take nanargmax.
+        axis (int): Along which axis to find the maximum. ``a`` is flattened by
+            default.
+    Returns:
+        cupy.ndarray: The indices of the maximum of ``a``
+        along an axis ignoring NaN values.
+    .. seealso:: :func:`numpy.nanargmax`
+    """
+    if a.dtype.kind in 'biu':
+        return argmin(a, axis=axis)
+
+    mask = cupy.isnan(a)
+
+    if cupy.any(cupy.all(mask, axis=axis)):
+        raise ValueError('All-NaN slice encountered')
```

Comment:
This error handle seems to degrade performance significantly. I discussed with another member offline and decided not to check this error. Could you remove this?

Reply:
Hey, apologies for the late response. I have removed the if condition for `ValueError` now. But I suppose `cupy.isnan(a)` is still required in `return argmin(cupy.where(cupy.isnan(a), cupy.inf, a), axis=axis)`

Reply:
Actually, I intended to comment only on L47. GitHub seems to display also last 3 lines...